components/ConvexClientProvider.tsx

```diff
@@ -1,16 +1,32 @@
 'use client';
 
-import { ReactNode, useCallback, useState } from 'react';
+import { ReactNode, useCallback, useState, useRef, useEffect } from 'react';
 import { ConvexReactClient } from 'convex/react';
 import { ConvexProviderWithAuth } from 'convex/react';
 import { AuthKitProvider, useAuth, useAccessToken } from '@workos-inc/authkit-nextjs/components';
 
+const noop = () => {};
+
+const isTokenValid = (token: string | undefined): boolean => {
+  if (!token) return false;
+
+  try {
+    const payload = JSON.parse(atob(token.split('.')[1]));
+    // Token is valid if it hasn't expired yet (with small buffer for clock skew)
+    return payload.exp * 1000 > Date.now() + 5000;
+  } catch {
+    return false;
+  }
+};
+
 export function ConvexClientProvider({ children }: { children: ReactNode }) {
   const [convex] = useState(() => {
-    return new ConvexReactClient(process.env.NEXT_PUBLIC_CONVEX_URL!);
+    const client = new ConvexReactClient(process.env.NEXT_PUBLIC_CONVEX_URL!);
+    return client;
   });
+
   return (
-    <AuthKitProvider>
+    <AuthKitProvider onSessionExpired={noop}>
```

Comment:
What does this do, disable some default onSessionExpired behavior? a comment would be helpful